core/templates/pages/learner-dashboard-page/learner-dashboard-icons.component.ts

```diff
@@ -0,0 +1,166 @@
+    if (this.learnerDashboardActivityIds) {
+      if (this.learnerDashboardActivityIds.includesActivity(
+        activityId)) {
+        return false;
+      } else {
+        if (this.isContainerNarrow) {
+          return true;
+        } else {
+          return this.activityIsCurrentlyHoveredOver;
+        }
+      }
+    }
+  }
+
+  belongsToLearnerPlaylist(): boolean {
+    var activityType = this.activityType;
+    if (this.learnerDashboardActivityIds) {
+      if (activityType === constants.ACTIVITY_TYPE_EXPLORATION) {
+        return (
+          this.learnerDashboardActivityIds.belongsToExplorationPlaylist(
+            this.activityId));
+      } else if (activityType === constants.ACTIVITY_TYPE_COLLECTION) {
+        return (
+          this.learnerDashboardActivityIds.belongsToCollectionPlaylist(
+            this.activityId));
+      }
+    }
+  }
+
+  belongsToCompletedActivities(): boolean {
+    var activityType = this.activityType;
+    if (this.learnerDashboardActivityIds) {
+      if (activityType === constants.ACTIVITY_TYPE_EXPLORATION) {
+        return (
+          this.learnerDashboardActivityIds.belongsToCompletedExplorations(
+            this.activityId));
+      } else if (activityType === constants.ACTIVITY_TYPE_COLLECTION) {
+        return (
+          this.learnerDashboardActivityIds.belongsToCompletedCollections(
+            this.activityId));
+      }
+    }
+  }
+
+  belongsToIncompleteActivities(): boolean {
+    var activityType = this.activityType;
+    if (this.learnerDashboardActivityIds) {
+      if (activityType === constants.ACTIVITY_TYPE_EXPLORATION) {
+        return (
+          this.learnerDashboardActivityIds.belongsToIncompleteExplorations(
+            this.activityId));
+      } else if (activityType === constants.ACTIVITY_TYPE_COLLECTION) {
+        return (
+          this.learnerDashboardActivityIds.belongsToIncompleteCollections(
+            this.activityId));
+      }
+    }
+  }
+
+  addToLearnerPlaylist(activityId: string, activityType: string): void {
+    var isSuccessfullyAdded = (
+      this.learnerPlaylistBackendApiService.addToLearnerPlaylist(
+        activityId, activityType));
+    if (isSuccessfullyAdded) {
+      if (activityType === constants.ACTIVITY_TYPE_EXPLORATION) {
+        this.learnerDashboardActivityIds.addToExplorationLearnerPlaylist(
+          activityId);
+      } else if (activityType === constants.ACTIVITY_TYPE_COLLECTION) {
+        this.learnerDashboardActivityIds.addToCollectionLearnerPlaylist(
+          activityId);
+      }
+      this.disablePlaylistTooltip();
+    }
+  }
+
+  removeFromLearnerPlaylist(
+      activityId: string, activityTitle: string, activityType: string): void {
+    this.learnerPlaylistBackendApiService.removeFromLearnerPlaylistModal(
+      activityId, activityTitle, activityType,
+      this.learnerDashboardActivityIds);
+  }
+}
+
+angular.module('oppia').directive(
+  'learnerDashboardIcons', downgradeComponent(
```

Comment:
change to oppiaLearnerDashboardIcons (same to other components)

Reply:
done!